lib/bap_disasm/bap_disasm_x86_lifter.ml

```diff
@@ -0,0 +1,2319 @@
+(** Native lifter of x86 instructions to the BAP IL *)
+
+open Core_kernel.Std
+open Bap_types.Std
+(* open Bil *)
+module BZ = Big_int_Z
+(* open Type *)
+(* open Arch *)
+(* open Var *)
+
+module BV = Bitvector
+
+module Util = struct
+  let id x = x (* useful for tabulating lists, for instance. *)
+
+  (* some of the "imported" zarith-processing functions use this. *)
+  (* XXX we don't like mutable state. Bitvector should be written so that the 2 big_int_Z
+   * functions that use this can be killed. *)
+
+  let index_ofq elt lst =
```

Comment:
`BAP` follows `Core` tradition of not throwing exceptions on occasional case. All functions that raise, a suffixed with `_exn` to denote this. We're using `Or_error` whenever the function can fail. Exceptions should be used only for programmer errors, not program errors.